.github/workflows/main.yml

```diff
@@ -4,279 +4,418 @@ name: hdf5 dev CI
 on:
   workflow_dispatch:
   push:
+  pull_request:
```

Comment:
Aside from the name, the 'on' key is the only difference between the two files.

This file has changed so much, it's probably better to look at the before and after individually rather than the diff.